sqladmin/models.py

```diff
@@ -662,3 +735,63 @@ class UserAdmin(ModelAdmin, model=User):
             self._column_labels.get(field, field.key) for field in search_fields
         ]
         return ", ".join(field_names)
+
+    def get_export_name(self, export_type: str) -> str:
+        """The file name when exporting."""
+        filename = "%s_%s.%s" % (
+            self.name,
+            time.strftime("%Y-%m-%d_%H-%M-%S"),
+            export_type,
+        )
+        return filename
+
+    def export_data(
+        self,
+        data: List[Any],
+        export_type: str = "csv",
+    ) -> StreamingResponse:
+        if export_type == "csv":
+            return self._export_csv(data)
+        else:
+            raise NotImplementedError("Only export_type='csv' is implemented.")
+
+    def _export_csv(
+        self,
+        data: List[Any],
+    ) -> StreamingResponse:
+        # https://docs.djangoproject.com/en/1.8/howto/outputting-csv/
+        class Echo(object):
```

Comment:
Maybe this can go into helpers.py?

Reply:
I tried to separate out as much of the API logic from the parsing logic with a function called `stream_to_csv`.